auacm/angular/src/app/pipes/limit-words.pipe.ts

```diff
@@ -7,18 +8,107 @@ import { Pipe, PipeTransform } from '@angular/core';
   name: 'limitWords'
 })
 export class LimitWordsPipe implements PipeTransform {
+  private converter: Converter = new Converter({'tables': true});
+
+  /**
+   * This transforms a Markdown string into HTML and shortens it based on the amount.
+   *
+   * @param words - the markdown text that needs to be shortened
+   * @param amount - the maximum word count of the returned string
+   */
   transform(words: string, amount: number): string {
+    const htmlWords = this.converter.makeHtml(words);
+    const parsedHtml = document.createElement( 'html' );
+    parsedHtml.innerHTML = htmlWords;
+
+    const tempChildren = parsedHtml.children[1].childNodes;
+    const children = [];
+    for (let i = 0; i < tempChildren.length; i++) {
+      children.push(tempChildren[i]);
+    }
+
+    // let finalHtml = document.createElement( 'html' );
     let finalString = '';
-    const allWords = words.split(' ');
-    let count = 0;
-    for (let i = 0; i < amount && i < allWords.length; i++) {
-      finalString += allWords[i] + ' ';
-      count++;
-    };
-    if (count === amount) {
-      return finalString.trim() + ' ...';
+    let size = 0;
+    for (let i = 0; i < children.length; i++) {
+      const element = this.getStringForElement(children[i], '', size, amount);
+      finalString += element.value;
```

Comment:
Is element.size unrelated to element.value? In this case it seems strange to be doing `+=` at one point, but doing a regular assignment to the other one.

I think a more readable approach would be:
```
let element = new CustomString('', 0);
for (let i = 0; i < children.length; i++) {
  element = this.getStringForElement(children[i],  element.value, element.size, amount);
}
return element.value;
```

I'm not 100% sure, but I think that should do the same thing, and to me it makes more sense.